src/scripts/main.js

```diff
@@ -1,3 +1,40 @@
 'use strict';
 
 // write code here
+const tree = document.querySelector('.tree');
+
+addSpanToFirstTextChild(tree);
+
+tree.addEventListener('click', (e) => {
+  const span = e.target.closest('span');
+  const li = span.closest('li');
+  const list = li.querySelector('ul');
+
+  if (list.style.display === 'none') {
+    list.style.display = '';
+  } else {
+    list.style.display = 'none';
+  }
+});
+
+function addSpanToFirstTextChild(list) {
+  const liList = list.querySelectorAll('li');
+
+  for (const li of liList) {
+    if (!li.firstChild) {
+      return;
```

Comment:
This check returns if the `li` has no first child. However, it might be more appropriate to continue to the next `li` instead of returning from the function entirely.